src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -709,24 +714,34 @@ private void setArgument(ArgumentDefinition argumentDefinition, List<String> val
     }
 
     /**
-     * Expand any collection values that are ".list" argument files, and add them
-     * to the list of values for that argument.
+     * Expand any collection value that references a filename that ends in one of the accepted expansion
+     * extensions, and add the contents of the file to the list of values for that argument.
+     * @param argDef the ArgumentDefinition being populated
      * @param originalValues
      * @return a list containing the original entries in {@code originalValues}, with any
      * values from list files expanded in place, preserving both the original list order and
      * the file order
      */
-    private List<String> expandListFile(final List<String> originalValues) {
-        List<String> expandedValues = new ArrayList<>(originalValues.size());
-        for (String stringValue: originalValues) {
-            if (stringValue.endsWith(COLLECTION_LIST_FILE_EXTENSION)) {
-                expandedValues.addAll(loadCollectionListFile(stringValue));
-            }
-            else {
-                expandedValues.add(stringValue);
+    private List<String> expandListFile(final ArgumentDefinition argDef, final List<String> originalValues) {
+        if (!argDef.suppressFileExpansion) {
```

Comment:
I like this change (although I prefer to being able to change the extension in a per-argument basis), but the configuration sounds much better for the change of the extensions itself. Have a look to the clean PR (I will comment with the number once it is here).